tests/unittests/tests-checksum/tests-checksum-crc8.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 Benjamin Valentin <benpicco@googlemail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdint.h>
+
+#include "embUnit/embUnit.h"
+
+#include "checksum/crc8.h"
+
+#include "tests-checksum.h"
+
+static void test_checksum_crc8_sequence_empty(void)
+{
+    unsigned char buf[] = "";
+    uint8_t expect = 0xFF;
+
+    TEST_ASSERT_EQUAL_INT(expect ,crc8(buf, sizeof(buf) - 1));
+}
+
+static void test_checksum_crc8_sequence_1a(void)
+{
+    unsigned char buf[] = "A";
+    uint8_t expect = 0xA0;
+
+    TEST_ASSERT_EQUAL_INT(expect, crc8(buf, sizeof(buf) - 1));
+}
+
+static void test_checksum_crc8_sequence_256a(void)
+{
+    unsigned char buf[] = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAA";
+    uint8_t expect = 0xF0;
```

Comment:
What's the source of these test values?

Reply:
The test cases are from `tests-checksum-crc16-ccitt.c`.
The values are just what the current crc8 implementation outputs for those buffers.

Reply:
Shouldn't this be at least be checked against an external implementation? E.g. http://tomeko.net/online_tools/crc8.php?lang=en

Reply:
I would use http://www.sunshine2k.de/coding/javascript/crc/crc_js.html which allows

- to select the CRC from a large list with predefined parameters or to define each parameter separately.
- to define the test string as string, byte sequence or bit sequence

With this calculator the expected values are correct for all tests.